tests/base/generic/test_inputlist.py

```diff
@@ -0,0 +1,346 @@
+                }
+        ], table_name = "input")
+        cls.pl["tail"] = InputList([2,4,8])
+
+        file_location = os.path.dirname(os.path.abspath(__file__))
+        pr = Project(file_location)
+        cls.file_name = os.path.join(file_location, "input.h5")
+        cls.hdf = ProjectHDFio(project=pr, file_name=cls.file_name,
+                               h5_path="/test", mode="a")
+
+    @classmethod
+    def tearDownClass(cls):
+        os.remove(cls.file_name)
+
+    ## Init tests
+    def test_init_none(self):
+        pl = InputList()
+        self.assertEqual(len(pl), 0, "not empty after initialized with None")
+
+    def test_init_list(self):
+        l = [1, 2, 3, 4]
+        pl = InputList(l)
+        self.assertEqual(len(pl), len(l),
+                "not the same length as source list")
+        self.assertEqual(list(pl.values()), l,
+                "conversion to list not the same as source list")
+
+    def test_init_tuple(self):
+        t = (1, 2, 3, 4)
+        pl = InputList(t)
+        self.assertEqual(len(pl), len(t),
+                "not the same length as source tuple")
+        self.assertEqual(tuple(pl.values()), t,
+                "conversion to tuple not the same as source tuple")
+
+    def test_init_set(self):
+        s = {1, 2, 3, 4}
+        pl = InputList(s)
+        self.assertEqual(len(pl), len(s),
+                "not the same length as source set")
+        self.assertEqual(set(pl.values()), s,
+                "conversion to set not the same as source set")
+
+    def test_init_dict(self):
+        d = {"foo": 23, "test case": "bar"}
+        pl = InputList(d)
+        self.assertEqual(tuple(pl.items()), tuple(d.items()),
+                "source dict items not preserved")
+        with self.assertRaises(ValueError,
+                               msg = "no ValueError on invalid initializer"):
+            pl = InputList({2: 0, 1: 1})
+
+
+    # access tests
+    def test_get_nested(self):
+        n = [
+                {"foo": "bar"},
+                2,
+                42,
+                {"next":
+                    [0,
+                        {"depth": 23}
+                    ]
+                }
+        ]
+        pl = InputList(n)
+        self.assertEqual(type(pl[0]), InputList,
+                "nested dict not converted to InputList")
+        self.assertEqual(type(pl["3/next"]), InputList,
+                "nested list not converted to InputList")
+        self.assertEqual(type(pl["0/foo"]), str,
+                "nested str converted to InputList")
+
+    def test_get_item(self):
+        self.assertEqual(self.pl[0], {"foo": "bar"},
+                "index with integer does not give correct element")
+        self.assertEqual(self.pl[0]["foo"], "bar",
+                "index with string does not give correct element")
+        with self.assertRaises(IndexError,
+                               msg = "no IndexError on out of bounds index"):
+            self.pl[15]
```

Comment:
```suggestion
            self.assertEqual(self.pl[15], 42)
```
Does this work? Then codacy would not complain

Reply:
I feel that this would hide the true intent of the test, checking whether invalid indices raise exceptions, how about `print()`ing it?

Reply:
print is fine for me